File: 백준/Bronze/2444. 별 찍기 － 7/별 찍기 － 7.swift
// [Int] N (immutable): 마름모의 대각선 길이 / 2 + 1 (1 <= N <= 100)
let N: Int = Int(readLine() ?? "1") ?? 1

// 위쪽 부분 출력 (중앙 포함)
for i in 0..<N {
    // 공백 출력
    print(String(repeating: " ", count: (N - i - 1)), terminator: "")
    // 별 출력
    print(String(repeating: "*", count: (2 * i + 1)))
}

// 아래쪽 부분 출력
for j in 0..<(N - 1) {
    // 공백 출력
    print(String(repeating: " ", count: (j + 1)), terminator: "")
    // 별 출력
    print(String(repeating: "*", count: (2 * (N - j - 2) + 1)))
}

File: 백준/Bronze/2444. 별 찍기 － 7/별 찍기 － 7.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

    private static String repeat(char ch, int count) {
        StringBuilder sb = new StringBuilder(count);
        for (int i = 0; i < count; i++) {
            sb.append(ch);
        }
        return sb.toString();
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N;
        try {
            N = Integer.parseInt(br.readLine());
        } catch (Exception e) {
            N = 1;
        }

        // 위쪽 (중앙 포함)
        for (int i = 0; i < N; i++) {
            System.out.print(repeat(' ', N - i - 1));
            System.out.println(repeat('*', 2 * i + 1));
        }

        // 아래쪽
        for (int j = 0; j < N - 1; j++) {
            System.out.print(repeat(' ', j + 1));
            System.out.println(repeat('*', 2 * (N - j - 2) + 1));
        }
    }
}